tools/FakeItEasy.Build/Program.cs

```diff
@@ -110,10 +111,16 @@ public static void Main(string[] args)
             RunTargets(args);
         }
 
-        private static void RunTests(string testDirectory)
+        private static void RunTests(string testDirectory, string framework)
         {
             var xml = Path.GetFullPath(Path.Combine(TestsDirectory, Path.GetFileName(testDirectory) + ".TestResults.xml"));
-            Run("dotnet", $"xunit -configuration Release -nologo -nobuild -noautoreporters -notrait \"explicit=yes\" -xml {xml}", testDirectory);
+            Run("dotnet", $"xunit -framework {framework} -configuration Release -nologo -nobuild -noautoreporters -notrait \"explicit=yes\" -xml {xml}", testDirectory);
         }
+
+        private static string[] GetFrameworks(string project) =>
+            File.ReadAllText($"{project}/{project.Split('/').Last()}.{(project.Contains("VB") ? "vbproj" : "csproj")}")
+                .Split("<TargetFrameworks>", 2)[1]
+                .Split("</TargetFrameworks>", 2)[0]
+                .Split(';');

```

Comment:
Ouch! This is probably a bit too much for an expression-bodied method :wink:
Making it a full method with intermediate variables would help readability.
Also, since an MSBuild project is XML, why not use Linq to XML?

Reply:
> Is it the path that bothers you?

It's the whole "do everything in a single statement", mostly. I can understand what it does, but it's hard to parse. Introducing variables makes everything more explicit IMO.


> However, instead of manipulating the XML, I'd probably try and find some Microsoft package which abstracts away the project entirely.

Well, Microsoft.Build could do it, but it's probably overkill. Something like this would do the trick:

```csharp
var doc = XDocument.Load(filename);
var frameworksElement = doc.Root.Elements("PropertyGroup").Elements("TargetFrameworks").First();
return frameworkElement.Value.Split(";");
```
Anyway, leave it as is if you prefer, I'll probably patch it later :wink: